### PR TITLE
Leveraged Staking L2 Strategy

### PR DESCRIPTION
**Pull request by larryob**
TODO

- [x] Lower bounds of check in `testMutateTotalLockedValue`
- [x] Complete other TODOs in code

### CODE REVIEW

File: src/strategies/LidoLevL2.sol [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        uint256 expectedWstETh = _ethToWstEth(ethBorrowed);
+        uint256 wstEth =
+            CURVE.exchange({x: uint256(0), y: 1, dx: ethBorrowed, min_dy: expectedWstETh.slippageDown(slippageBps)});
+
+        // Deposit wstETH in AAVE
+        asset.safeApprove(address(AAVE), type(uint256).max);
+        AAVE.deposit(address(WSTETH), wstEth, address(this), 0);
+
+        // Borrow 90% of wstETH value in ETH using e-mode
+        uint256 ethToBorrow = _wstEthToEth(wstEth).mulDivDown(8999, 10_000);
```
Comment (maruf0011) [inline src/strategies/LidoLevL2.sol:104]: Do we need to borrow fixed ratio? 
We can just borrow required amount like `ethBorrowed - asset.balanceOf(this)`. Won't have any idle assets.

File: src/strategies/LidoLevL2.sol [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+    /// @notice The leverage factor of the position multiplied by 100. E.g. 150 would be 1.5x leverage.
+    uint256 public leverage = 992; // 9.92x
+
+    function setLeverage(uint256 _leverage) external onlyRole(STRATEGIST_ROLE) {
+        leverage = _leverage;
+    }
+
+    /// @notice The percentage of the collateral to borrow from AAVE in bps.
+    uint256 public borrowBps = 8999; // 89.99%
```
Comment (maruf0011) [inline src/strategies/LidoLevL2.sol:226]: is it required? No usage.

**Review by maruf0011 — APPROVED** [review]
LGTM :) 
Have some questions.

**Comment by maruf0011** [comment]
If we change the leverage, how do we change the leverage of existing investment?

File: src/strategies/LidoLevL2.sol (hunk 2 above)
Comment (larryob) [reply src/strategies/LidoLevL2.sol:226]: Good catch. This is used to determine how much wETH to borrow during deposits.

File: src/strategies/LidoLevL2.sol (hunk 1 above)
Comment (larryob) [reply src/strategies/LidoLevL2.sol]: Great comment! We must always borrow exactly enough to repay the flashloan. Fixed.

**Pull request merged**